ftp/1.6/modules/ROOT/pages/ftp-dynamic-connection.adoc

```diff
@@ -0,0 +1,149 @@
+
+. In the *Mule Palette* view, select *HTTP > Listener*.
+. Drag *HTTP Listener* to the Studio canvas.
+. On the *HTTP Listener* configuration screen, optionally change the value of the *Display Name* field.
+. Set *Path* to `/multitenant`.
+. Click the plus sign (*+*) next to the *Connector configuration* field to configure a global element that can be used by all instances of HTTP Listener in the app.
+. On the *General* tab, set the following fields:
++
+* *Host*: `All Interfaces [0.0.0.0] (default)`
+* *Port*: `8081`
++
+. Click *OK*.
+. Drag a *Set Variable* component to the right of the *HTTP Listener* source.
+. Set *Name* to `content` and *Value* to `#[payload]`.
+. Drag the File Connector *Read* operation to the right of the *Set Variable* component.
+. Set *File Path* to `recipients.csv`. This reads a CSV file that contains a random set of FTP destinations with columns such as `host`, `user`, and `port`.
+. Click the plus sign (*+*) next to the *Connector configuration* field to configure a global element that can be used by all instances of File Connector in the app.
+. Accept the default settings and click *OK*.
+. On the *MIME Type* tab of the *Read* operation, for *MIME Type*, select *application/csv*.
+
+. In Studio, drag a *Transform Message* component to the right of the *Read* operation.
+. In the *Output* section of the component, add the following DataWeave code to map the columns of the CSV file:
++
+.DataWeave script:
+[source,dataweave,linenums]
+----
+%dw 2.0
+output application/java
+---
+payload map using (row = $) {
+   host: row.Host,
+   user: row.User,
+   password: row.Password}
+----
++
+. Drag a *For Each* component to the right of the *Transform Message* component.
+. Drag an FTP *Write* operation into the *For Each* component. +
+On each For Each iteration, the *Write* operation writes the contents into each of the FTP destinations.
+. Click the plus sign (*+*) next to the *Connector configuration* field to configure a global element that can be used by all instances of FTP *Write* operation in the app, and set the following fields:
++
+* *Host*: `#[payload.host]`
+* *Username*: `#[payload.username]`
+* *Password*: `#[payload.password]`
++
+[start=6]
+. Click *OK*.
+. Set *Path* to `demo.txt`.
+. Set the *Content* field to `payload`.
+. Drag a *Set Payload* component to the right of the *For Each* component.
+. Set the *Value* field to `Multicast Ok`.
+
+== Run and Test Your Mule Application
+
+After creating your Mule application, run and test it:
+
+. In Studio, save your Mule app.
+. Click the project name in *Package Explorer* and then click *Run* > *Run As* > *Mule Application*.
+. Open a browser and type `+http://0.0.0.0:8081/multitenant+`. +
+
+Note about Mule 4 behavior for Mule 3 users:
+
+* Although the information posted through the HTTP Listener source is written to each FTP site multiple times, the repeatable streams feature in Mule 4 prevents the overhead of consuming the stream multiple times.
```

Comment:
Change "of" to "from" toward the end of the sentence so that it reads: "...Mule 4 prevents the overhead from consuming the stream multiple times."